openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/azureSQLConnection.json

```diff
@@ -59,6 +63,43 @@
       "type": "string",
       "default": "ODBC Driver 18 for SQL Server"
     },
+    "authenticationMode": {
+      "title": "Authentication Mode",
+      "description": "This parameter determines the mode of authentication for connecting to AzureSQL using ODBC. If 'Active Directory Password' is selected, you need to provide the password. If 'Active Directory Integrated' is selected, password is not required as it uses the logged-in user's credentials. This mode is useful for establishing secure and seamless connections with AzureSQL.",
+      "properties": {
+        "authentication": {
+          "title": "Authentication",
+          "description": "Authentication from Connection String for AzureSQL.",
+          "type": "string",
+          "enum": [
+            "ActiveDirectoryIntegrated",
+            "ActiveDirectoryPassword"
+          ]
+        },
+        "encrypt": {
+          "title": "Encrypt",
+          "description": "Encrypt from Connection String for AzureSQL.",
+          "type": "boolean"
+        },
+        "trustServerCertificate": {
+          "title": "Trust Server Certificate",
+          "description": "Trust Server Certificate from Connection String for AzureSQL.",
+          "type": "boolean"
+        },
+        "connectionTimeout": {
+          "title": "Connection Timeout",
+          "description": "Connection Timeout from Connection String for AzureSQL.",
+          "type": "integer",
+          "default": 30
+        }
+      },
+      "required": [
+        "Encrypt",
+        "Trust Server Certificate",
```

Comment:
should the required be the title or the name?

Reply:
ahh, right, my bad.. but i tried with the required using name, i think it's not working as expected, the Authentication if not provided with anything, should be optional, if we add ActiveDirectory related type, then it should have other inputs added by the user, removing the same..
<img width="884" alt="Screenshot 2024-03-21 at 9 32 50 PM" src="https://github.com/open-metadata/OpenMetadata/assets/40225091/8e78aeff-60df-4afe-8578-0eaff10e7323">
As all the params will be available from azure itself, the users will have to copy paste it here.. and some of the params are not needed somewhere